libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/RoadShieldContentManager.kt

```diff
@@ -0,0 +1,193 @@
+package com.mapbox.navigation.ui.maneuver
+
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.ui.maneuver.model.Component
+import com.mapbox.navigation.ui.maneuver.model.Maneuver
+import com.mapbox.navigation.ui.maneuver.model.RoadShield
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldComponentNode
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldError
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldResult
+import com.mapbox.navigation.utils.internal.LoggerProvider
+import com.mapbox.navigation.utils.internal.ThreadController
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+internal class RoadShieldContentManager {
```

Comment:
To consider for the future we should revisit the implementation here and use `Flow` instead.

Reply:
cc @LukasPaczos

Reply:
I actually did use a `channel` implementation first but it wasn't working well for me. I looked at `flow`s as well and couldn't find the tools I needed.

Reply:
Would love to hear about what tools you couldn't find. But again, we can discuss this separately 👍